examples/scripts/script_reslc.py

```diff
@@ -0,0 +1,143 @@
+#     processes=1,  # Number of Python processes per worker
+#     walltime="3:00:00",  # Reserve each worker for X hour
+#     scheduler_options={"dashboard_address": f":{FREE_SOCKET}"},  # Host Dashboard in a free socket
+# )
+# logger.info(f"Dask dashboard hosted at port: {FREE_SOCKET}.")
+# logger.info(
+#     f"If you are forwarding Jupyter Server to a local port 8889, \
+#     you can access it at: localhost:8889/proxy/{FREE_SOCKET}/status"
+# )
+
+# Option 2: Intiate a new SLURMCluster
+# Uncomment the following part to give an exist Dask SLURMCluster
+ADDRESS = "tcp://XX.X.X.XX:12345" # Manual input: Dask schedular address
+SOCKET = 12345 # Manual input: port number. It should be the number after ":" of ADDRESS
+cluster = None  # Keep this None, needed for an if statement
+logger.info(f"Dask dashboard hosted at port: {SOCKET}.")
+logger.info(
+    f"If you are forwarding Jupyter Server to a local port 8889, \
+    you can access it at: localhost:8889/proxy/{SOCKET}/status"
+)
+
+if __name__ == "__main__":
+    logger.info("Initializing ...")
+
+    if cluster is None:
+        # Use existing cluster
+        client = Client(ADDRESS)
+    else:
+        # Scale a certain number workers
+        # each worker will appear as a Slurm job
+        cluster.scale(jobs=N_WORKERS)
+        client = Client(cluster)
+
+    # Make figure directory if not exists
+    path_figure.mkdir(exist_ok=True)
+
+    logger.info("Loading data ...")
+    # Metadata
+    f_mother_res = mother_dir / "slave.res"
+    metadata = read_metadata(f_mother_res)
+
+    # Coordinates
+    f_lam = mother_dir / "lam.raw"
+    f_phi = mother_dir / "phi.raw"
+
+    # Mother SLC
+    f_mother_slc = mother_dir / "slave_rsmp_reramped.raw"
+
+    # List of SLC
+    f_ifgs = list(sorted(stack_dir.rglob("2*/cint_srd.raw")))
+    f_ifgs = f_ifgs[:3]
+
+    shape = (metadata["n_lines"], metadata["n_pixels"])
+    dtype_slc_ifg = np.dtype([("re", np.float32), ("im", np.float32)])
+    dtype_lam_phi = np.float32
+
+    # Lazy loading mother SLC and ifg stack
+    mother = sarxarray.from_binary([f_mother_slc], shape, dtype=dtype_slc_ifg, chunks=reading_chunks)
+    ifgs = sarxarray.from_binary(f_ifgs, shape, dtype=dtype_slc_ifg, chunks=reading_chunks)
+
+    # Generate reconstructed SLCs
+    slc_recon = intf_to_slc(mother, ifgs)
+
+    # Extract real and image part. remove other fields. convert to float16
+    slc_recon_output = slc_recon.copy()
+    slc_recon_output = slc_recon_output.assign(
+        {
+            "real": slc_recon_output["complex"].real.astype(np.float16),
+            "imag": slc_recon_output["complex"].imag.astype(np.float16),
+        }
+    )
+    slc_recon_output = slc_recon_output.drop_vars(["complex", "amplitude", "phase"])
+
+    # Rechunk and write as zarr
+    slc_recon_output = slc_recon_output.chunk({"azimuth": writing_chunks[0], "range": writing_chunks[1]})
+    if overwrite_zarr:
+        slc_recon.to_zarr("nl_veenweiden_s1_dsc_t037.zarr", mode="w")
```

Comment:
Hi Freek,
as we discussed on Tuesday, I have put these two action points in issue #22, since they need test on Spider. I will work on them after finishing issue #10 .
For now the script only output non-mother SLCs in `np.float16`, with real and imag part separate. Just to confirm, as we discussed on Tuesday, there is no `cpxfloat16` available in numpy (I think you mean `np.complex32`? any how neither exists in numpy) so we are going to store real and imag part separately, both in `np.float16`.

Reply:
Hi Ou,
Sorry, I should be more clear in my writing. With cpxfloat16 I meant the 'concept' of storing the complex data in float16 format, indeed the real and imaginary part separately. So as you described.